src/Analyzer/Resolve/evaluateScalarSubqueryIfNeeded.cpp

```diff
@@ -131,6 +132,7 @@ void QueryAnalyzer::evaluateScalarSubqueryIfNeeded(QueryTreeNodePtr & node, Iden
         /// When execute `INSERT INTO t WITH ... SELECT ...`, it may lead to `Unknown columns`
         /// exception with this settings enabled(https://github.com/ClickHouse/ClickHouse/issues/52494).
         subquery_settings[Setting::use_structure_from_insertion_table_in_table_functions] = false;
+        subquery_settings[Setting::allow_experimental_parallel_reading_from_replicas] = 0;
```

Comment:
I still can't quite get why this is needed. Even if parallel replicas need to be disabled, `QueryTreePassManager` runs passes a few lines later.

Reply:
The reason - https://github.com/ClickHouse/ClickHouse/pull/88923/files#r2481608862

Do you think the referenced query failure with PR should be addressed differently?